nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/storage/sql/TestCloudFrontRedirect.java

```diff
@@ -0,0 +1,101 @@
+import static org.junit.Assume.assumeTrue;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.DISTRIB_DOMAIN_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.PRIVATE_KEY_ID_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.CloudFrontBinaryManager.PRIVATE_KEY_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_ID_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_SECRET_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_NAME_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_PREFIX_PROPERTY;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * Test ignored as it requires lots of file and tokens. Manually executed
+ */
+@Ignore
+@RunWith(FeaturesRunner.class)
+@Features(RuntimeFeature.class)
+public class TestCloudFrontRedirect extends AbstractS3BinaryTest<CloudFrontBinaryManager> {
+    protected static Map<String, String> PROPERTIES = Collections.emptyMap();
+
+    @BeforeClass
+    public static void beforeClass() {
+        PROPERTIES = new HashMap<>();
+        // NOTE THAT THE TESTS WILL REMOVE ALL FILES IN THE BUCKET!!!
+        // ********** NEVER COMMIT THE SECRET KEYS !!! **********
+        String bucketName = "CHANGETHIS";
+        String idKey = "CHANGETHIS";
+        String secretKey = "CHANGETHIS";
+
+        String privateKeyPath = "CHANGETHIS";
+        String privateKeyId = "CHANGETHIS";
+        String distribution = "CHANGETHIS";
+        // ********** NEVER COMMIT THE SECRET KEYS !!! **********
+
+        PROPERTIES.put(BUCKET_NAME_PROPERTY, bucketName);
+        PROPERTIES.put(BUCKET_PREFIX_PROPERTY, "testfolder/");
+        PROPERTIES.put(AWS_ID_PROPERTY, idKey);
+        PROPERTIES.put(AWS_SECRET_PROPERTY, secretKey);
+        PROPERTIES.put(DISTRIB_DOMAIN_PROPERTY, distribution);
+        PROPERTIES.put(PRIVATE_KEY_PROPERTY, privateKeyPath);
+        PROPERTIES.put(PRIVATE_KEY_ID_PROPERTY, privateKeyId);
+        boolean disabled = bucketName.equals("CHANGETHIS");
+        assumeTrue("No AWS credentials configured", !disabled);
+    }
+
+    @Before
+    public void tearUp() throws IOException {
+        removeObjects();
+    }
+
+    @Test
+    public void testRemoteUri() throws Exception {
+        testStoreFile();
+        String key = listObjects().stream().findAny().orElseThrow(Exception::new);
+
+        URI remoteUri = getBinaryManager().getRemoteUri(key, null, null);
+        assertNotNull(remoteUri);
+        System.out.println(String.format("curl -v \"%s\"", remoteUri));
```

Comment:
Why?
I agree to not have any `System.out` in an automatic test as it shows nothing; and it could also hide interesting informations. But, in this test case, it helps a lot to check headers between the and system / CF.
But yes... those headers can also be checked in the test 👍